Mathlib/Order/CompletePartialOrder.lean

```diff
@@ -0,0 +1,243 @@
+/-
+Copyright (c) 2023 Christopher Hoskin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Christopher Hoskin
+-/
+import Mathlib.Order.Directed
+import Mathlib.Order.OmegaCompletePartialOrder
+import Mathlib.Data.Set.Finite
+import Mathlib.Data.Finset.Basic
+
+/-!
+# Complete Partial Order
+
+This file begins by showing that each set `s` in a join-semilattice generates a directed set `d`
+such that:
+
+- `s` is a subset of `d`
+- `s` and `d` share the same upper bounds
+- `u` is the least upper bound of `s` if and only if it is the least upper bound of `d`
```

Comment:
Surely that third sentence follows from the second?

Reply:
Yes, I've removed the second.